packages/SwingSet/src/kernel/vatAdmin/vatAdminWrapper.js

```diff
@@ -65,22 +68,27 @@ export function buildRootObject(vatPowers) {
     }
   }
 
-  // the kernel sends this when the vat halts
-  function vatTerminated(vatID, error) {
-    // 'error' is undefined if adminNode.terminate() killed it, else it
-    // will be a RangeError from a metering fault
-    const { resolve, reject } = running.get(vatID);
-    running.delete(vatID);
-    if (error) {
+  // the kernel sends this when the vat halts intentionally
```

Comment:
Another parallel construction that should probably be unified.

Reply:
How's this?

Reply:
That's cleaner, though I was actually more concerned about the implementations of `terminateVat`/`terminateVatOnFailure` in `kernel.js`

Reply:
I refactored there, too. The only additional thing I can think of to do there is to pass in a pair of strings, or actually construct the messages based on which case it is, both of which seemed ugly to me.  Can you see further improvements?